### ISSUE TITLE
Function Parameter Names

### ISSUE THREAD

**Issue by daviejaneway** [harlanhaskins/LLVMSwift#139]
Hello again, hope you are well.

Is it possible to have parameter names in functions?

e.g.

```
define i32 @foo(%i32, %i32) {
entry:
  ret i32 1
}
```

Is there a way to assign names to the params when creating a FunctionType? If not, could you offer any advice on how best to reference arguments within the function body please?

Many thanks.

**Comment by CodaFi** [comment]
Hi @daviejaneway,

Apologies for not seeing this earlier.  I take it from your self-closure that you found the answer but I wanted to follow up anyways.

To answer your question, LLVM doesn't let you express named parameters in the IR.  One way you can remedy this on your end - especially if you have a language that gives names to lvalues, is to carry around a dictionary mapping named values to IRValues with each FunctionType you create.  But you won't be able to serialize that mapping, at least not until we get debug info support with the LLVM 7.0 release.

**Comment by daviejaneway** [comment]
Hi @CodaFi,

Really appreciate the reply. I did indeed find the solution to my issue, it turned out a bug in my project was leading me down the wrong path of thinking I needed named params. Turns out I don't!

Again, massive thanks for your work on this project. I have just hit a stage in my own language (Orbit) implementation where I can build & run trivial programs. LLVMSwift has played a huge part in getting to that stage.
